apps/server/src/modules/provisioning/strategy/sanis/schulconnex-response-mapper.ts

```diff
@@ -172,6 +182,57 @@ export class SchulconnexResponseMapper {
 		return mapped;
 	}
 
+	private static mapGroupDuration(duration: SchulconnexLaufzeitResponse | undefined): TimePeriode | undefined {
+		if (!duration) {
+			return undefined;
+		}
+
+		if (duration.von && duration.bis && duration.bis > duration.von) {
+			return {
+				from: duration.von,
+				until: duration.bis,
+			};
+		}
+
+		if (duration.vonlernperiode && duration.bislernperiode) {
+			const fromPeriode: TimePeriode | undefined = SchulconnexResponseMapper.mapLernperiode(duration.vonlernperiode);
+			const untilPeriode: TimePeriode | undefined = SchulconnexResponseMapper.mapLernperiode(duration.bislernperiode);
+
+			if (!fromPeriode || !untilPeriode || fromPeriode > untilPeriode) {
+				return undefined;
+			}
+
+			return {
+				from: fromPeriode.from,
+				until: untilPeriode.until,
+			};
+		}
+
+		return undefined;
+	}
+
+	public static mapLernperiode(lernperiode: string): TimePeriode | undefined {
```

Comment:
Can be private or?

Reply:
I made it public so I can test it separately in the unit test, since it contains some logic that should be carefully tested.

Reply:
Okay right.. protected only works for inheritance, right?
I would like to avoid to make something public just to for testing purpose.